src/tools_for_rules.jl

```diff
@@ -256,31 +254,31 @@ function increment_and_get_rdata!(f, r, t::CRC.Thunk)
     return increment_and_get_rdata!(f, r, CRC.unthunk(t))
 end
 
-@doc"""
-    rrule_wrapper(f::CoDual, args::CoDual...)
+@doc """
+     rrule_wrapper(f::CoDual, args::CoDual...)
 
-Used to implement `rrule!!`s via `ChainRulesCore.rrule`.
+ Used to implement `rrule!!`s via `ChainRulesCore.rrule`.
 
-Given a function `foo`, argument types `arg_types`, and a method of `ChainRulesCore.rrule`
-which applies to these, you can make use of this function as follows:
-```julia
-Mooncake.@is_primitive DefaultCtx Tuple{typeof(foo), arg_types...}
-function Mooncake.rrule!!(f::CoDual{typeof(foo)}, args::CoDual...)
-    return rrule_wrapper(f, args...)
-end
-```
-Assumes that methods of `to_cr_tangent` and `to_mooncake_tangent` are defined such that you
-can convert between the different representations of tangents that Mooncake and
-ChainRulesCore expect.
+ Given a function `foo`, argument types `arg_types`, and a method of `ChainRulesCore.rrule`
+ which applies to these, you can make use of this function as follows:
+ ```julia
+ Mooncake.@is_primitive DefaultCtx Tuple{typeof(foo), arg_types...}
+ function Mooncake.rrule!!(f::CoDual{typeof(foo)}, args::CoDual...)
+     return rrule_wrapper(f, args...)
+ end
+ ```
+ Assumes that methods of `to_cr_tangent` and `to_mooncake_tangent` are defined such that you
+ can convert between the different representations of tangents that Mooncake and
+ ChainRulesCore expect.
 
-Furthermore, it is _essential_ that
-1. `f(args)` does not mutate `f` or `args`, and
-2. the result of `f(args)` does not alias any data stored in `f` or `args`.
+ Furthermore, it is _essential_ that
+ 1. `f(args)` does not mutate `f` or `args`, and
+ 2. the result of `f(args)` does not alias any data stored in `f` or `args`.
```

Comment:
This looks like a bug to me. Does it also look like a bug to you @gdalle ?

Reply:
Perhaps, I've never used `@doc` in my own code so I've never encountered it. In any case, JuliaFormatter is the ground truth, so this might be grounds for opening an issue there. Not much we can do from here, except if the style file lets us skip the `@doc` docstrings.

Reply:
To be honest, I think I'm probably misusing the `@doc` macro here -- I believe that we can just remove it and use a regular docstring.

Reply:
let's fix that in a separate PR?

Reply:
Makes sense. Let's get this merged.